alpaca/rest.go

```diff
@@ -329,10 +332,36 @@ func (c *Client) CloseAllPositions(req CloseAllPositionsRequest) ([]Order, error
 		return nil, err
 	}
 
-	var orders orderSlice
-	if err = unmarshal(resp, &orders); err != nil {
+	var closeAllPositions closeAllPositionsSlice
+	if err = unmarshal(resp, &closeAllPositions); err != nil {
 		return nil, err
 	}
+
+	var (
+		orders = make([]Order, 0, len(closeAllPositions))
+		errs   = make([]error, 0, len(closeAllPositions))
+	)
+	for _, capr := range closeAllPositions {
+		if capr.Status == http.StatusOK {
+			var order Order
+			if err := easyjson.Unmarshal(capr.Body, &order); err != nil {
+				return nil, err
+			}
+			orders = append(orders, order)
+			continue
+		}
+		var apiErr APIError
+		if err := easyjson.Unmarshal(capr.Body, &apiErr); err != nil {
+			return nil, err
+		}
+		apiErr.StatusCode = capr.Status
+		errs = append(errs, &apiErr)
+	}
+
+	if len(errs) > 0 {
+		return orders, errors.Join(errs...)
```

Comment:
minor: you can always return this as I believe errors.Join() will return nil if the slice doesn't contain any errors